pkg/config/nodetreemodel/missing_node.go

```diff
@@ -15,7 +15,13 @@ import (
 // missingLeafImpl is a none-object representing when a child node is missing
 type missingLeafImpl struct{}
 
-var missingLeaf missingLeafImpl
+var _ Node = (*missingLeafImpl)(nil)
+
+var missingLeaf *missingLeafImpl
```

Comment:
Could you add a comment stating that this is intentionally nil, and methods do not dereference it.